### PR TITLE
Userland: Implement recursive `rm`

### PR DESCRIPTION
**Pull request by christopherdumas**
Solves #161, by adding an option to recursively remove directories (and the files inside them).

### CODE REVIEW

File: Userland/rm.cpp [hunk 1]
```diff
@@ -1,18 +1,62 @@
...
-    if (argc != 2) {
-        fprintf(stderr, "usage: rm <path>\n");
-        return 1;
-    }
-    int rc = unlink(argv[1]);
-    if (rc < 0) {
-        perror("unlink");
+    struct stat path_stat;
+    int s = stat(path, &path_stat);
+    if (s == -1)
```
Comment (awesomekling) [inline Userland/rm.cpp]: We should also `perror("stat")` here so the user learns what the problem was.

File: Userland/rm.cpp [hunk 2]
```diff
@@ -1,18 +1,62 @@
...
+
+    if (S_ISDIR(path_stat.st_mode) && recursive) {
+        DIR* derp = opendir(path);
+        if (!derp) {
+            return 1;
+        }
+
+        while (auto* de = readdir(derp)) {
+            if (strcmp(de->d_name, ".") != 0 && strcmp(de->d_name, "..") != 0) {
+                char strbuf[256];
```
Comment (awesomekling) [inline Userland/rm.cpp]: This is a pretty small buffer. Use a `StringBuilder` instead and you don't have to worry about things fitting :)

```cpp
StringBuilder builder;
builder.append(path);
builder.append('/');
builder.append(de->d_name);
int s = remove(true, builder.to_string().characters());
...
```

**Comment by christopherdumas** [comment]
Okay, I made and tested those changes.

**Comment by awesomekling** [comment]
This looks very nice! Thanks for implementing it Christopher :)

**Comment by rburchell** [comment]
Great indeed :)

For extra credit, I think it would be nice to move this functionality "upstream" to CDirIterator: give it a new flag to allow recursion.

I can imagine this being functionality we'd want elsewhere over time -- but if nobody else does it, I'll take a look at it in the future.

**Comment by awesomekling** [comment]
Yeah that'd absolutely be great. Anytime we can use and/or improve LibCore idioms in Userland, there's a gold star sticker reward :)

**Pull request merged**